enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+---
+title: bpfd-on-ocp
+authors:
+  - "@anfredette"
+  - "@astoycos"
+  - "@Billy99"
+  - "@dave-tucker"
+reviewers:
+  - "@dhellmann"
+  - "@bparees"
+  - "@deads2k"
+  - "@msherif1234"
+  - "@jotak"
+  - "@erthalion"
+  - "@danwinship"
+approvers:
+  - "@knobunc"
+api-approvers: 
+  - "@knobunc"
+creation-date: 2023-02-21
+last-updated: 2023-03-14
+tracking-link:
+see-also:
+replaces:
+superseded-by:
+---
+
+# BPF Support for OCP with bpfd
+
+## Summary
+
+This enhancement proposes the following:
+- That the bpfd operator be included in OCP as a foundational operator.
+- That the bpfd operator be used by internal Red Hat OCP applications and
+  components for the loading and management of required BPF programs.
+- That bpfd be recommended and supported for use by OCP partners and users to
+  load and manage their BPF programs when running on OCP.
```

Comment:
While the use cases outside of core Red Hat applications are possible, it could help simplify the discussion to focus on the very specific needs of today, which are Red Hat components only.

I think that core networking + ACS are very nice illustrative examples. One is a core feature of OpenShift that needs this, and another is an add-on that would also make use of it.

Getting into letting end users *also* access it is another layer of complexity for the discussion. It may also be that supporting our own add-ons (ACS) is no different than solving it for a more general case from the bpfd perspective, but it seems like they have different support implications at least.

Reply:
Those are good points @russellb.  I had assumed that supporting our own features would be step one and then after that is working well, we would consider extending the API to customers.  I can reword the proposal to reflect this.

Reply:
> I believe this has to be a core feature.
We may not have the terminology right.  The point is that it has to be available and running before any of the mentioned features are enabled.  And, the number of features/components that use eBPF is expected to continue to grow.  Would it be more appropriate to call it a "core feature"?

Reply:
re: "core feature," I think what is in the text is fine. Maybe I was trying to respond to past discussion about whether it could be an optional add-on and I was saying "no, it can't." It's fine as-is for me.